compiler/testData/codegen/bytecodeText/intrinsicsCompare/byteSmartCast_after.kt

```diff
@@ -13,9 +11,19 @@ fun less4(a: Byte?, b: Byte?) = if (a is Byte && b is Byte) a < b else true
 
 fun less5(a: Any?, b: Any?) = if (a is Byte && b is Byte) a < b else true
 
+// JVM_TEMPLATES
 // 3 Intrinsics\.areEqual
 // 0 Intrinsics\.compare
 // 4 INVOKEVIRTUAL java/lang/Byte\.byteValue \(\)B
 // 2 INVOKEVIRTUAL java/lang/Number\.intValue \(\)I
 // 0 IFGE
 // 3 IF_ICMPGE
+
+// JVM_IR_TEMPLATES
```

Comment:
The change here is that instead of `Number.intValue`, `Number.byteValue` is used when known to be a byte. Additionally, the IR backend actually generates 3 `areEqual` calls, but the asm optimization method visitor then ends up unboxing and using `Number.byteValue` on the arguments.